### PR TITLE
fix kubelet panic in syncTerminatedPod

### LINKED ISSUE #103625: Kubelet panic in syncTerminatedPod
#### What happened:

When testing #102344 against OpenShift CI, I discovered a panic in the kubelet: https://github.com/openshift/kubernetes/pull/851#issuecomment-877469501

Further investigation revealed that it is possible for `syncTerminatedPod` to panic under certain circumstances: https://github.com/kubernetes/kubernetes/blob/a0d77e5763ac04ddc713bf2e284580d32fd4b079/pkg/kubelet/pod_workers.go#L685-L713

We hit this case when `update.Options.RunningPod != nil` (podStatus remains nil) and `update.WorkType == TerminatedPodWork` (nil will be passed along to `syncTerminatedPod`). I'm not sure why exactly that would happen but it seems possible.

#### What you expected to happen:

Don't panic!

#### How to reproduce it (as minimally and precisely as possible):

Unclear, I've been trying to trace the code/find evidence of this happening in k8s CI but haven't found a match yet. (https://github.com/kubernetes/test-infra/issues/22851 would be nice :smile:)

#### Anything else we need to know?:

/sig node
/cc @rphillips @smarterclayton 

/priority important-soon
/milestone v1.22

#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by CaoDonghui123**
#### What type of PR is this?

/kind bug
#### Which issue(s) this PR fixes:

Fixes #103625

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1832,6 +1832,9 @@ func (kl *Kubelet) syncTerminatedPod(ctx context.Context, pod *v1.Pod, podStatus
 
 	// generate the final status of the pod
 	// TODO: should we simply fold this into TerminatePod? that would give a single pod update
+	if nil == podStatus {
```
Comment (endocrimes) [inline pkg/kubelet/kubelet.go:1835]: please reverse this (`podStatus == nil`)

Comment (smarterclayton) [reply pkg/kubelet/kubelet.go:1835]: Hold on.  Running pods shouldn't be terminated.  So it's a bug to get into this path.

/hold

Comment (endocrimes) [reply pkg/kubelet/kubelet.go:1835]: Yeah I wasn't sure about this - (trying to figure out how it could happen rn)

**Comment by smarterclayton** [comment]
The fix isn't in this method (running pods should never go into terminated, and pod status is a required argument for syncTerminatedPod).

See https://github.com/kubernetes/kubernetes/issues/103625#issuecomment-878294492

This would be the wrong fix because we don't want a runningPod to ever make it out of syncTerminatingPod (runtime only pods have no config and are cleaned up by housekeeping, not the pod worker).

**Comment by ehashman** [comment]
@CaoDonghui123 thank you for the patch, unfortunately this fix is not so simple :) The podStatus shouldn't have been nil at this point to begin with, so we will need to dig deeper.

/close

**Pull request closed without merging**